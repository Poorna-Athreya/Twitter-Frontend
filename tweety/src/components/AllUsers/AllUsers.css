 .users-container{
    margin-top:1%;
    display:flex;
    flex-flow: row wrap;
    gap:15px;
    padding: 3%;
    padding-top:0%;
} 
@media screen and (max-width: 300px) {
    .users-container{
        margin-top: 3%;
        margin-bottom: 3%;
    }
}
