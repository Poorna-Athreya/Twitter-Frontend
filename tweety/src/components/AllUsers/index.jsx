/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import { ALL_USERS } from '../../constants/users';
import UserCard from '../UserCard';
import './AllUsers.css';
import makeRequest from '../../utils/makeRequest';

function AllUsers() {
  const [allUsers, setAllUsers] = useState(ALL_USERS);
  // setAllUsers(ALL_USERS);

  // const [isInitialised, setIsInitialised] = useState(false);
  const allUsersDisplay = allUsers.map((eachUser) => (
    <UserCard
      userName={eachUser.name}
      key={eachUser.id}
      id={eachUser.id}
      location={eachUser.location}
      userHandle={eachUser.handle}
    />
  ));
  return (
    <div className="users-container">
      {allUsersDisplay}
    </div>
  );
}
export default AllUsers;
