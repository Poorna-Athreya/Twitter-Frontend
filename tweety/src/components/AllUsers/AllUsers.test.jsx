import { render, screen } from '@testing-library/react';
import React from 'react';
import AllUsers from './index';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({ useNavigate: () => mockNavigate }));

const mockAllUsers = [
  {
    id: 110,
    name: 'Twitter Dev',
    handle: 'TwitterDev',
    location: 'Internet',
  },
  {
    id: 111,
    name: 'Lorem Ipsum',
    handle: 'LoremIpsum',
    location: 'Internet',
  }];

describe('AllUsers', () => {
  it('should take a snapshot for AllUsers', () => {
    const { asFragment } = render(<AllUsers allUsers={mockAllUsers} />);
    expect(asFragment(<AllUsers allUsers={mockAllUsers} />)).toMatchSnapshot();
  });
  it('should render all the user cards when AllUsers is rendered', async () => {
    render(<AllUsers allUsers={mockAllUsers} />);
    expect(await screen.findAllByText('Twitter Dev')).toBeTruthy();
    expect(await screen.findAllByText('@TwitterDev')).toBeTruthy();
    expect(await screen.findAllByText('Internet')).toBeTruthy();
    expect(await screen.findAllByText('Lorem Ipsum')).toBeTruthy();
    expect(await screen.findAllByText('@LoremIpsum')).toBeTruthy();
  });
});
