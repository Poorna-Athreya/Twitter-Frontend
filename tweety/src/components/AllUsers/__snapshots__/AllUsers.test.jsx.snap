// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllUsers should take a snapshot for AllUsers 1`] = `
<DocumentFragment>
  <div
    class="users-container"
    data-testid="userContainer"
  >
    <button
      aria-label="user-card-button"
      class="user-card"
      data-testid="userCardButton"
      type="button"
    >
      <h1
        class="user-name"
        data-testid="userCardName"
      >
        Twitter Dev
      </h1>
      <br />
      <div
        class="user-details"
      >
        <p
          class="user-handle"
          data-testid="userCardHandle"
        >
          @TwitterDev
        </p>
        <p
          class="user-location"
          data-testid="userCardLocation"
        >
          Internet
        </p>
      </div>
    </button>
    <button
      aria-label="user-card-button"
      class="user-card"
      data-testid="userCardButton"
      type="button"
    >
      <h1
        class="user-name"
        data-testid="userCardName"
      >
        Lorem Ipsum
      </h1>
      <br />
      <div
        class="user-details"
      >
        <p
          class="user-handle"
          data-testid="userCardHandle"
        >
          @LoremIpsum
        </p>
        <p
          class="user-location"
          data-testid="userCardLocation"
        >
          Internet
        </p>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
